File: dictionary/src/main/java/ru/romanow/gateway/services/SeriesService.java
package ru.romanow.gateway.services;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Service;
import ru.romanow.gateway.domain.SeriesEntity;
import ru.romanow.gateway.mappers.ModelMapper;
import ru.romanow.gateway.models.Series;

@Service
public class SeriesService
        extends BaseCrudService<String, SeriesEntity, Series> {

    public SeriesService(
            JpaRepository<SeriesEntity, String> repository,
            ModelMapper<String, SeriesEntity, Series> mapper
    ) {
        super(repository, mapper);
    }

}

File: dictionary/src/main/java/ru/romanow/dictionary/services/SeriesService.kt
package ru.romanow.dictionary.services

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Service
import ru.romanow.dictionary.domain.SeriesEntity
import ru.romanow.dictionary.models.Series

@Service
class SeriesService(repository: JpaRepository<SeriesEntity, String>) :
    BaseCrudService<String, SeriesEntity, Series>(repository) {

    override fun toModel(entity: SeriesEntity) =
        Series(
            name = entity.name,
            type = entity.type?.name,
            complexity = entity.complexity?.name,
            age = entity.age
        )
}
